### PR TITLE
D2 ecs

### PR DESCRIPTION
**Pull request by bschofield-va**
Adds 
- ECS deployments
- Integration testing
- Dockerhub support
- Simulated failures

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -84,6 +67,10 @@ pipeline {
     choice(name: 'VPC', choices: ["QA", "UAT", "Staging", "Production", "Staging-Lab", "Lab" ],
       description: "Environment to deploy into")
     string(name: 'PRODUCT', defaultValue: 'none', description: "The product to deploy.")
+    choice(name: 'PROMOTION', choices: ["auto", "none" ],
+      description: "Environment to deploy into")
```
Comment (jhulbert-va) [inline Jenkinsfile:71]: This is the same description as `VPC` could get confusing.

File: Jenkinsfile [hunk 2]
```diff
@@ -84,6 +67,10 @@ pipeline {
...
+    choice(name: 'SIMULATED_FAILURE', choices: [ "none","activate","initialize","validate","before-deploy-green","deploy-green","verify-green","switch-to-blue","verify-blue","after-verify-blue","finalize","before-rollback","rollback","verify-rollback","after-rollback" ],
+      description: "Environment to deploy into")
```
Comment (jhulbert-va) [inline Jenkinsfile:73]: Samesies. Copy Pasta!

File: Jenkinsfile [hunk 3]
```diff
@@ -92,10 +79,11 @@ pipeline {
         script {
           // Sometimes these parameters are not defaulted... thanks Jenkins.
           if (env.DEBUG == null) { env.DEBUG='false' }
-          if (env.DEPLOYER_VERSION == null) { env.DEPLOYER_VERSION='latest' }
+          if (env.DEPLOYER_VERSION == null) { env.DEPLOYER_VERSION='mvn-3.6-jdk-14' }
```
Comment (jhulbert-va) [inline Jenkinsfile:82]: Why not `latest`? Deployments should all be containerized and java/maven agnostic, right?

File: Jenkinsfile [hunk 4]
```diff
@@ -92,10 +79,11 @@ pipeline {
...
           if (env.PRODUCT == null) { env.PRODUCT='none' }
+          if (env.PROMOTION == null) { env.PROMOTION='auto' }
           if (env.VPC == null) { env.VPC='QA' }
-          if (env.GIT_BRANCH != 'd2') {
+          if (env.GIT_BRANCH != 'd2' && env.GIT_BRANCH != 'd2-ecs') {
```
Comment (jhulbert-va) [inline Jenkinsfile:86]: Should just be d2 before merging.

File: environments/lab.conf [hunk 5]
```diff
@@ -1,7 +1,11 @@
-export CLUSTER_ID=l25
-export CLUSTER_SSH_KEY="$KUBERNETES_LAB_SSH_KEY"
+export GREEN_LOAD_BALANCER_PROTOCOL=https
```
Comment (jhulbert-va) [inline environments/lab.conf:1]: I thought this was going to be `DEPLOYMENT_TEST_*` 
[Relevant Conversation](https://libertyits.slack.com/archives/GBULYN18X/p1604682910035300?thread_ts=1604681788.032200&cid=GBULYN18X)

File: plugins/it [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  return 0;
+}
+
+on-initialize() {
+  loadTestConf
+  echo "Pulling test image $TEST_IMAGE"
+  docker pull -q $TEST_IMAGE
+}
+
+on-verify-green() {
```
Comment (jhulbert-va) [inline plugins/it:24]: Are we not doing `verify-blue` with smoke test(s)?

File: bin/alb [hunk 7]
```diff
@@ -0,0 +1,378 @@
...
+
+tmpFile() {
+  local f=$(mktemp $@)
+  DELETE_FILES+=( $f )
+  echo $f
+}
+
+requireParam() {
+  local param="${1}"
+  local value="${2:-}"
```
Comment (jhulbert-va) [inline bin/alb:114]: If you wanted to, `$(eval echo $"$param")` would work and you wouldn't need to specify the second param. Might be a bit fragile though if the string has other bash sensitive characters in it?

File: Jenkinsfile (hunk 3 above)
Comment (bschofield-va) [reply Jenkinsfile:82]: I wanted to be more specific here so I didn't accidentally pick a jdk 15 instance when we start on that

File: Jenkinsfile (hunk 4 above)
Comment (bschofield-va) [reply Jenkinsfile:86]: it should and I will remove the d2-ecs bit _after_

File: plugins/it (hunk 6 above)
Comment (bschofield-va) [reply plugins/it:24]: we should, I just haven't done it yet

File: environments/lab.conf (hunk 5 above)
Comment (bschofield-va) [reply environments/lab.conf:1]: ^ this is configuration for the environment.... not variables passed to the integration test container

The container does get `DEPLOYMENT_TEST_*` variables when it's ran

**Pull request merged**